x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -1155,25 +1155,31 @@ private SearchRequest buildQueryToUpdateDestinationIndex() {
 
     /**
      * Indicates if an audit message should be written when onFinish is called for the given checkpoint
-     * We audit the first checkpoint, and then every 10 checkpoints until completedCheckpoint == 99
-     * Then we audit every 100, until completedCheckpoint == 999
+     * We audit every checkpoint for the first 10 checkpoints until completedCheckpoint = 9.
+     * Then we audit every 10 checkpoints until completedCheckpoint == 99.
+     * Then we audit every 100, until completedCheckpoint == 999.
      *
-     * Then we always audit every 1_000 checkpoints
+     * Then we always audit every 1_000 checkpoints.
      *
      * @param completedCheckpoint The checkpoint that was just completed
      * @return {@code true} if an audit message should be written
      */
     protected boolean shouldAuditOnFinish(long completedCheckpoint) {
-        if (++logCount % logEvery != 0) {
-            return false;
-        }
-        if (completedCheckpoint == 0) {
-            return true;
-        }
-        int log10Checkpoint = (int) Math.floor(Math.log10(completedCheckpoint));
-        logEvery = log10Checkpoint >= 3 ? 1_000 : (int) Math.pow(10.0, log10Checkpoint);
-        logCount = 0;
-        return true;
+        return remainingCheckpointsToSkipAuditing.getAndUpdate(count -> {
```

Comment:
Nice idea!
This way we achieve atomicity of the update operation but also improve performance as we do not perform heavy mathematical operations.

Reply:
Thank you!